modules/plugins/languages/arduino.nix

```
@@ -0,0 +1,129 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.generators) mkLuaInline;
+  inherit (lib.lists) optionals;
+  inherit (lib.meta) getExe getExe';
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.types) enum listOf nullOr package str;
+  inherit (lib.nvim.attrsets) mapListToAttrs;
+  inherit (lib.nvim.types) mkGrammarOption;
+
+  cfg = config.vim.languages.arduino;
+
+  defaultServers = ["arduino-language-server"];
+  servers = {
+    arduino-language-server = {
+      enable = true;
+      cmd =
+        [
+          (getExe pkgs.arduino-language-server)
+          "-clangd"
+          (
+            if cfg.lsp.clangdPackage == null
+            then "clangd"
+            else getExe' cfg.lsp.clangdPackage "clangd"
+          )
+          "-cli"
+          (
+            if cfg.lsp.cliPackage == null
+            then "arduino-cli"
+            else getExe cfg.lsp.cliPackage
+          )
+          "-cli-config"
+          cfg.lsp.cliConfigPath
+          "-fqbn"
+          cfg.lsp.fqbn
+        ]
+        ++ cfg.lsp.extraArgs;
```

```suggestion
      cmd =
        [
          (getExe pkgs.arduino-language-server)
          "-clangd"
          (getExe' pkgs.clang-tools "clangd")
          "-cli"
          (getExe pkgs.arduino-cli)
          "-cli-config"
          "$HOME/.arduino15/arduino-cli.yaml"
          "-fqbn"
          cfg.lsp.fqbn
        ];
```

@NotAShelf @Soliprem I need a second opinion, should `cfg.lsp.clangdPackage` and friends be added here, or should we remove them and let users configure via `vim.lsp.servers`?

since `cfg.lsp.fqbn` is required, I think we'll need to keep it regardless if the others get removed